gitCoreJGit/src/main/java/com/virtuslab/gitcore/impl/jgit/GitCoreRepository.java

```diff
@@ -130,13 +137,35 @@ private ObjectId gitCoreCommitToObjectId(IGitCoreCommit commit) throws GitCoreEx
     } catch (IOException e) {
       throw new GitCoreException("Cannot get current branch", e);
     }
+
     if (ref == null) {
       throw new GitCoreException("Error occurred while getting current branch ref");
     }
+
+    String currentBranchName = null;
+
     if (ref.isSymbolic()) {
-      String currentBranchName = Repository.shortenRefName(ref.getTarget().getName());
+      currentBranchName = Repository.shortenRefName(ref.getTarget().getName());
+    } else {
+      Path headNamePath = jgitRepo.getDirectory().toPath().resolve("rebase-apply").resolve("head-name");
+      if (!headNamePath.toFile().isFile()) {
+        headNamePath = jgitRepo.getDirectory().toPath().resolve("rebase-merge").resolve("head-name");
```

Comment:
Hmm wait but do we have a guarantee that `.git/rebase-merge/head-name` exists? What if we're just in detached HEAD state and not rebasing?

Reply:
Good spot! Fixed.